### PR TITLE
Add Matter Product Development Best Practices guide

### PR DESCRIPTION
**Pull request by stingaci**
- Integrates Product Development Best Practices document under guides
- Adheres to Connectivity Standards Alliance (Alliance) naming and acronym rules
- Updates English guides index
- Adds new acronyms and proper nouns to Vale dictionaries

Requested Label: guides

### CODE REVIEW

File: src/en/guides/product-development-best-practices.md [hunk 1]
```diff
@@ -0,0 +1,238 @@
...
+**General Product Development Activities:**
+
+* **Hardware Bring Up:** Fabricate and assemble form-factor Printed Circuit Board Assemblies (PCBAs). Perform basic power-on tests, interface validation, and initial Radio Frequency (RF) circuit testing for Wi-Fi/Thread, and BLE.
+* **Base Firmware:** Integrate initial firmware and conduct software unit/integration tests on the hardware.
+* **Internal Field Testing:** Ensure internal team members interact with the working prototype for early feedback.
+* **Updated Risk Assessment:** Leveraging testing data & user feedback, identify and document new key risks in hardware, software, mechanics, and network performance. Develop mitigation plans.
+
+**Matter Specific Activities:**
+
+* **Core Connectivity Testing:** Verify successful Matter commissioning onto a test fabric. Confirm the device appears correctly with its defined Matter device type and clusters. Verify basic on/off or primary function control via Matter.
```
Comment (robszewczyk) [inline src/en/guides/product-development-best-practices.md:112]:
```suggestion
* **Core Connectivity Testing:** Verify successful Matter commissioning onto a test fabric -- test certificates may be used. Confirm the device appears correctly with its defined Matter device type and clusters. Verify basic on/off or primary function control via Matter.
```

File: src/en/guides/product-development-best-practices.md [hunk 2]
```diff
@@ -0,0 +1,238 @@
...
+* **Network Joining:** Verify successful joining to a test Ethernet/Wi-Fi/Thread network and test basic network reachability.
```
Comment (robszewczyk) [inline src/en/guides/product-development-best-practices.md:113]: This is  emphasizing low level connectivity check, a precursor to Matter connectivity outlined above. Swap order with the previous bullet

File: src/en/guides/product-development-best-practices.md [hunk 3]
```diff
@@ -0,0 +1,238 @@
...
+* **Manufacturing Test Prep:** Develop and validate factory test stations and software.
+* **Field Testing (General):** Launch an internal field testing program to gather qualitative and quantitative feedback on usability and feature performance.
+
+**Matter Specific Activities:**
+
+* **Matter Self-Testing & Pre-Certification:** [Matter self-testing](https://handbook.buildwithmatter.com/certification/certifying-a-product/self-pre-test/) is critical at this stage. Validate spec compliance using the [YAML](https://github.com/project-chip/connectedhomeip/tree/master/src/app/tests/suites/certification) or [Python](https://github.com/project-chip/connectedhomeip/tree/master/src/python_testing) SDK folders. Run manual verification steps from the [official release documentation](https://groups.csa-iot.org/wg/members-all/document/folder/2269). Test with the [Matter Test Harness](https://github.com/project-chip/certification-tool/blob/main/docs/Matter_TH_User_Guide/Matter_TH_User_Guide.adoc), as this mimics ATL evaluation.
+* **In-Depth Protocol Testing:** Conduct commissioning stress tests, multi-admin commissioning scenarios, subscription testing, and group control testing.
```
Comment (robszewczyk) [inline src/en/guides/product-development-best-practices.md:138]:
```suggestion
* **In-Depth Protocol Testing:** Conduct commissioning stress tests, multi-admin commissioning scenarios, subscription testing, and group control testing. Test Device Authentication Certificates may be used.
```

File: src/en/guides/product-development-best-practices.md [hunk 4]
```diff
@@ -0,0 +1,238 @@
...
+
+**General Product Development Activities:**
+
+* **Manufacturing & Reliability:** Build units on the mass production line. Monitor and optimize factory yield. Conduct extensive Highly Accelerated Life Testing (HALT) and environmental stress tests.
+* **Regressions:** Execute full regression test suites on hardware and software to catch issues introduced by process changes.
+* **Field Testing Expansion:** Expand the field testing program participation with DVT units by at least 2x to focus on long-term stability and edge cases.
+
+**Matter Specific Activities:**
+
+* **Formal Certification:** Submit DVT units to Authorized Test Laboratories (ATLs) for formal certifications (Regulatory, Safety, Wi-Fi, Bluetooth, Thread, and Matter). Apply for Interoperability testing simultaneously with Matter certification to discover and address issues with popular smart home ecosystems.
```
Comment (robszewczyk) [inline src/en/guides/product-development-best-practices.md:169]: @cecille what's the required certificate path here?  It seems to me, that you should be able to run through the ATL certification with test certs rather than the prod certs, but I'd like to confirm

File: src/en/guides/product-development-best-practices.md [hunk 5]
```diff
@@ -0,0 +1,238 @@
...
+* **Large-Scale Interoperability:** Expand interoperability testing with a wider range of Matter controllers and networking infrastructure (Wi-Fi Routers, Thread Border Routers). Participate in Alliance Test Events.
```
Comment (robszewczyk) [inline src/en/guides/product-development-best-practices.md:170]:
```suggestion
* **Large-Scale Interoperability:** Expand interoperability testing with a wider range of Matter controllers and networking infrastructure (Wi-Fi Routers, Thread Border Routers). 
```

Test Events are used in the specification development.  The equivalent for released clusters might be plugfests, but those are generally ad hoc, and thus should not be on the product launch path. Interop testing with Alliance facilities is already called out in the previous bullet.

File: src/en/guides/product-development-best-practices.md [hunk 6]
```diff
@@ -0,0 +1,238 @@
...
+
+**General Product Development Activities:**
+
+* **Production Line Validation:** Run a significant build on the finalized assembly lines to verify repeatable and scalable processes.
+* **Test & Packaging:** Verify all factory test stations are operating correctly. Verify final packaging, labeling, and accessories.
+* **Factory Shipped Image (FSI) Lock:** Finalize and lock the software image to be flashed on devices during production.
+
+**Matter Specific Activities:**
+
+* **Provisioning Validation:** Ensure the FSI correctly includes all necessary firmware, Matter configurations, and pre-loaded credentials. Validate that the factory lines are successfully and securely injecting individual DACs, discriminators, and PAKE verifiers, and that physical QR codes match the digital records perfectly.
```
Comment (robszewczyk) [inline src/en/guides/product-development-best-practices.md:194]:
```suggestion
* **Provisioning Validation:** Ensure the FSI correctly includes all necessary firmware, Matter configurations, and pre-loaded credentials. Validate that the factory lines are successfully and securely injecting individual DACs, discriminators, and PAKE verifiers, and that physical QR codes match the digital records perfectly.  Production DACs must be used at this stage.
```

**Review by robszewczyk — APPROVED** [review]
I don't want to be the blocker and I will be out over the next bit.  The document reads harsh.  Tracking the product quality (in the last bullet) is challenging given the initial pitch (accept the tradeoff of not having the single app) and a potential lack of standardized access to fleet metrics across the ecosystems.

**Review by p0fi — APPROVED** [review]
I think it’s fine to add it as is, despite the valid comments @robszewczyk made. I think the content is fine, and we’ll incorporate it into a structural rework of the handbook that I’m hopefully getting to soon.

File: src/en/guides/product-development-best-practices.md (hunk 2 above)
Comment (stingaci) [reply src/en/guides/product-development-best-practices.md:112]: Swapped them

File: src/en/guides/product-development-best-practices.md (hunk 4 above)
Comment (stingaci) [reply src/en/guides/product-development-best-practices.md:169]: @cecille gentle ping on this question

Comment (cecille) [reply src/en/guides/product-development-best-practices.md]: Yeah, you do need to use test certs. In many cases the PKI provider won't give production certs until the CD is granted. But that's really between the manufacturer and their PKI provider or the internal security people running their CA.

For re-cert, using prod certs is expected and allowed, but the tests themselves don't care either way.

The inteorp lab asks for production devices, as do many of the works with programs, though they will take pre-release for folks looking to get badging.
